### PR TITLE
New filters and display paper count

### PR DESCRIPTION
**Pull request by tylerhutcherson**
New category filter (multi tag select) option and dynamic "searchable" papers count. Cleaner index/query management classes for us to abstract later. 

Addresses:
- https://github.com/RedisVentures/redis-arXiv-search/issues/8
- https://github.com/RedisVentures/redis-arXiv-search/issues/7

### CODE REVIEW

File: backend/vecsim_app/api/routes.py [hunk 1]
```diff
@@ -1,31 +1,43 @@
...
+from vecsim_app.search_index import SearchIndex
 
 
 paper_router = r = APIRouter()
 redis_client = redis.from_url(config.REDIS_URL)
+embeddings = Embeddings()
+search_index = SearchIndex()
 
-async def process_paper(p):
+async def process_paper(p, i: int) -> dict:
```
Comment (Spartee) [inline backend/vecsim_app/api/routes.py:22]: Should use ``t.Dict`` with proper typing. i.e. ``t.Dict[str, str]`` or something like that

**Review by Spartee — COMMENTED** [review]
Leaving comments for now, let me know what you think.

curious to know why you changed over to the category tag and not the full name.

File: backend/vecsim_app/embeddings.py [hunk 2]
```diff
@@ -3,45 +3,42 @@
...
-model = SentenceTransformer('sentence-transformers/all-mpnet-base-v2')
 
-# String cleaner
-def clean_description(description: str):
-    if not description:
-        return ""
-    # remove unicode characters
-    description = description.encode('ascii', 'ignore').decode()
+class Embeddings:
+    model = SentenceTransformer('sentence-transformers/all-mpnet-base-v2')
```
[the remark this answers is not in the document]
Comment (tylerhutcherson) [reply backend/vecsim_app/embeddings.py:8]: good idea

File: frontend/src/views/Home.tsx [hunk 3]
```diff
@@ -56,150 +60,26 @@ export const Home = (props: Props) => {
...
-  //   'stat.AP': 'Applications',
-  //   'stat.CO': 'Computation',
-  //   'stat.ME': 'Methodology',
-  //   'stat.ML': 'Machine Learning',
-  //   'stat.OT': 'Other Statistics',
-  //   'stat.TH': 'Statistics Theory'
-  // };
-
+  const categoryOptions = [
+    'cs.LG',
```
[the remark this answers is not in the document]
Comment (tylerhutcherson) [reply frontend/src/views/Home.tsx:64]: Agreed. The category tag is what we have in the dataset natively and that's what is currently indexed in the document data model in redis. We should be able to maintain a small map of name -> tag here in the front end to do the conversion

**Pull request merged**